### PR TITLE
Frustum camera module

### PR DESCRIPTION
**Pull request by Dedado3000**

### CODE REVIEW

File: Source/Modules/ModuleDebugDraw.cpp [hunk 1]
```diff
@@ -627,6 +627,21 @@ void ModuleDebugDraw::DrawBoundingBox(const AABB& aabb)
     if(showBoundingBoxes) dd::aabb(aabb.minPoint, aabb.maxPoint, dd::colors::Orange);
 }
 
+void ModuleDebugDraw::DrawBoundingBox(const OBB& obb)
+{
+    if (showBoundingBoxes) {
```
Comment (xavi1111) [inline Source/Modules/ModuleDebugDraw.cpp:632]: You have to change the opening bracket to adapt the coding convention

File: Source/DataModels/3DModels/Model.cpp [hunk 2]
```diff
@@ -175,21 +176,18 @@ const float4x4& Model::GetRotationF4x4() const
 
 void Model::SetScale(const float3& scale)
 {
-	if (scale.x == 0 || scale.y == 0 || scale.z == 0) return;
-	aabb.Scale(aabb.CenterPoint(), float3(scale.x / this->scale.x, scale.y / this->scale.y, scale.z / this->scale.z));
 	this->scale = scale;
-	obb = aabb.Transform(float4x4::FromTRS(float3{ 0,0,0 }, GetRotationF4x4(), scale));
+	obb = aabb.Transform(float4x4::FromTRS(translation, GetRotationF4x4(), scale));
```
Comment (OriolSors) [inline Source/DataModels/3DModels/Model.cpp:180]: We will need to refine this for generating the OBB properly

File: Source/Modules/ModuleDebugDraw.cpp [hunk 3]
```diff
@@ -627,6 +627,21 @@ void ModuleDebugDraw::DrawBoundingBox(const AABB& aabb)
...
+        ddVec3 points[8];
+        obb.GetCornerPoints(points);
+        ddVec3 orderedPoints[8] = { points[0], points[1], points[3], points[2], points[4], points[5], points[7], points[6] };
+        dd::box(orderedPoints, dd::colors::Orange);
+    }
```
Comment (OriolSors) [inline Source/Modules/ModuleDebugDraw.cpp:638]: We will need to ensure this method works properly when OBB well loaded

File: Source/Modules/ModuleScene.cpp [hunk 4]
```diff
@@ -13,10 +13,15 @@ ModuleScene::~ModuleScene()
...
 {
 	root = new GameObject("Root");
-
 	return true;
 }
 
+bool ModuleScene::IsInsideACamera(const OBB& obb)
+{
+	// TODO: We have to add all the cameras in the future
+	return false;
```
Comment (OriolSors) [inline Source/Modules/ModuleScene.cpp:22]: TODO for the next sprint

**Review by OriolSors — APPROVED** [review]
Reviewed after refactor

**Review by xavi1111 — APPROVED** [review]
Good work we will need to work on the improvements Oriol pointed out on the next sprint.

**Pull request merged**